project1/mceuropeanengine.hpp

```diff
@@ -60,10 +67,72 @@ namespace QuantLib {
              Size requiredSamples,
              Real requiredTolerance,
              Size maxSamples,
-             BigNatural seed);
+             BigNatural seed,
+             bool constant); //! définition du constructeur de la classe MCEuropeanEngine_2 avec en paramètre ajout du booléen
+             
       protected:
         boost::shared_ptr<path_pricer_type> pathPricer() const;
+       
+      private: 
+        bool constant_; //! définition de l'attribut boolean
+        boost::shared_ptr<GeneralizedBlackScholesProcess> process_;
+        bool antithetic_;
+        Size steps_, stepsPerYear_, samples_, maxSamples_;
+        Real tolerance_;
+        bool brownianBridge_;
+        BigNatural seed_;        
+        
+      public: //! méthode pathGenerator() reprise de la classe MCVanillaEngine
+        boost::shared_ptr<path_generator_type> pathGenerator() const {
+			Size dimensions = process_->factors();
+			TimeGrid grid = this->timeGrid();
+			typename RNG::rsg_type generator =
+					RNG::make_sequence_generator(dimensions*(grid.size()-1),seed_);
+			if (this->constant_ ){
+				
+				
+				return boost::shared_ptr<path_generator_type>(
+					new path_generator_type(process_, grid,
+											generator, brownianBridge_));
+			}
+			// we can not compile this part
+		
+			else{
+				boost::shared_ptr<GeneralizedBlackScholesProcess> process =boost::dynamic_pointer_cast<GeneralizedBlackScholesProcess>(this->process_);
+				boost::shared_ptr<PlainVanillaPayoff> payoff = boost::dynamic_pointer_cast<PlainVanillaPayoff>(this->arguments_.payoff);
+                QL_REQUIRE(payoff, "non-plain payoff given");
+				/*boost::shared_ptr<constantBlackScholesProcess> const_process =boost::dynamic_pointer_cast<constantBlackScholesProcess>(
+                                                        new constantBlackScholesProcess(process->stateVariable(), 
+                                                        this->arguments_.exercise->lastDate(),
+                                                        payoff->strike(), process->riskFreeRate(),
+                                                                process->blackVolatility(), 
+                                                                process->dividendYield(),
+                                                                boost::shared_ptr<StochasticProcess1D::discretization>(new EulerDiscretization)));*/
+                                //
+                boost::shared_ptr<StochasticProcess1D::discretization> disc = new EulerDiscretization ;
```

Comment:
```suggestion
                boost::shared_ptr<StochasticProcess1D::discretization> disc(new EulerDiscretization);
```
(some compilers don't accept your way of writing it).
An alternative would be to give a default discretization to the `constantBlackScholesProcess` ( see the comment on this in the other header).  This way, you won't even need to pass it here.